test/test_app.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/python
+# coding=utf-8
+
+import pytest
+import requests
+
+
+@pytest.mark.parametrize('endpoint,expected', [('/mnist/4', '{"label": 4, "predicted": 4}'),
+                                               ('/mnist/4432', '{"label": 0, "predicted": 0}'),
+                                               ('/mnist/10000',
+                                                'Invoke using the form /mnist/<index of test image>. Index must be in '
+                                                'the range [0:9999]. For example, /mnist/24'),
+                                               ('/mnist',
+                                                'Not found. Invoke using the form /mnist/<index of test image>. For '
+                                                'example, /mnist/24'),
+                                               ('/mni',
+                                                'This service verifies a model using the MNIST Test data set. Invoke '
+                                                'using the form /mnist/<index of test image>. For example, /mnist/24'),
+                                               ('/mnist/test',
+                                                'Not found. Invoke using the form /mnist/<index of test image>. For '
+                                                'example, /mnist/24')])
+def test_request_response(get_free_port, thread, endpoint, expected):
```

Comment:
The thread parameter isn't used.

```suggestion
def test_request_response(get_free_port, endpoint, expected):
```

Reply:
it is not called directly, but it is needed for the test to run - ie the threads need to be set up